yapapi/services.py

```diff
@@ -880,3 +880,18 @@ def stop(self):
         """Signal the whole :class:`Cluster` to stop."""
         for s in self.instances:
             self.stop_instance(s)
+
+    @property
+    def has_active_instances(self) -> bool:
+        """Check if this :class:`Cluster` has any instances that are currently active.
+
+        Specifically, this includes any instances that are starting, running or stopping - or -
+        in other words, the instances for which the Activities on the provider's end are still
+        available.
+        """
+        return any(s.is_available for s in self.instances)
+
+    @property
+    def has_starting_instances(self) -> bool:
+        """Check if this :class:`Cluster` has any instances that are currently starting."""
+        return any(s.state == ServiceState.starting for s in self.instances)
```

Comment:
Those are some adhoc functions useful from the examples POV.
Maybe we could make this more general? E.g.

```
def instances_with_state(self, *states) -> List[Service]:
    states = [str(state) for state in states]  # both `running` and `ServiceState.running` works
    return [i for i in self.instances if str(i.state) in states]
```
and than:
```
if cluster.instances_with_state('starting')  # instead of `has_starting_instances`

if cluster.instances_with_state(*ServiceState.available)  # instead of `has_active_instances`
```

(I'd look for a better name than `instances_with_state`)

Reply:
I prefer to have functions that are _actually_ useful rather than, again, over-verbose generalized ones

Reply:
Your `actually useful` is from the POV of examples we have. As far as I understand we're creating a general library that should serve a bilion purposes, not just blender and simple_service.

I think that some random `has_starting_instances` is just a really bad design. We'll add `has_terminated_instances` when we need it for some other example? Or maybe, idk, `has_exactly_3_stopping_instances` if one day this will be _actually_ useful?

If you don't like my generalized version, let's just drop this for now, after #560 is done all those things will get easier and we'll rethink this again.

Reply:
In general, I think we should **think much more** about such generalizations and stop doing things that are useful from the POV of our examples.

Reply:
I don't think "has_starting_instances" is just as bad as "has_exactly_3_stopping_instances"  ... otherwise, I do see your point but I think I'm going a bit above "pov of the examples" when thinking about the API than what you're suggesting, Janek...

Reply:
I don't see how `has_starting_instances` is above "pov of the examples", e.g. I would expect `has_running_instances` to be more useful.

But whatever, I still think this is really ugly, this is confusing for the user ("There is `has_starting_instances` and not `has_running_instances`, I wonder what is the reason? Because there is a reason for sure?"), this just makes `yapapi` a little-bit-worse.

IOW, I will not accept this. If you're not convinced than let's discuss this with the team.

Reply:
okay, I have reverted the addition of those methods